test/1.4/lib/T_signal_templates.py

```diff
@@ -0,0 +1,32 @@
+# © 2022 Intel Corporation
+# SPDX-License-Identifier: MPL-2.0
+
+import stest
+import simics
+simics.SIM_run_command('log-level 2')
+obj.port.insig.iface.signal.signal_raise()
+stest.expect_equal(obj.port.insig.signal_high, True)
+with stest.expect_log_mgr(log_type="spec-viol", obj=obj.port.insig, regex="already high"):
+    obj.port.insig.iface.signal.signal_raise()
```

Comment:
if you want to test all aspects of init/SIM_is_restoring_state logic, then you can consider saying `/// INSTANTIATE-MANUALLY` in the DML file and do `SIM_create_object` here.

Reply:
Thanks for the tip, added tests for init and checkpoint load.